docs/BYTESTRING.md

```diff
@@ -0,0 +1,78 @@
+---
+title: The ByteString Situation
```

Comment:
In order to avoid duplication for `Text` you could try to combine the two into one doc

Reply:
`Text` is not broken in the same way. ```genUnchecked `suchThat` isValid``` works for `Text`.

Reply:
Oh, cool. I haven't looked at the code, I just assumed so from our previous conversations